gst/nnstreamer/tensor_common.c

```diff
@@ -1259,19 +1260,21 @@ gst_gen_tensors_from_collectpad (GstCollectPads * collect,
     }
 
     if (GST_IS_BUFFER (buf)) {
-      mem = gst_buffer_get_memory (buf, 0);
-      gst_buffer_append_memory (tensors_buf, mem);
+      g_assert (gst_buffer_n_memory (buf) == in_configs.info.num_tensors);
+
+      for (i = 0; i < gst_buffer_n_memory (buf); ++i) {
+        mem = gst_buffer_get_memory (buf, i);
+        gst_buffer_append_memory (tensors_buf, mem);
```

Comment:
if counting is larger than 16, gst_buffer_append_memory() will merge the existing memory blocks to add new mem block.